app/tests/unit/services/objectPermission.spec.js

```diff
@@ -0,0 +1,94 @@
+const { MockModel, MockTransaction } = require('../../common/dbHelper');
+const { NIL: SYSTEM_USER } = require('uuid');
+
+jest.mock('../../../src/db/models/tables/objectPermission', () => MockModel);
+jest.mock('../../../src/db/models/tables/bucketPermission', () => MockModel);
+
+const { Permissions } = require('../../../src/components/constants');
+
+const service = require('../../../src/services/objectPermission');
+
+const bucketId = '00000000-0000-0000-0000-000000000000';
+const objectId = '00000000-0000-0000-0000-000000000000';
+const userId = '00000000-0000-0000-0000-000000000000';
+
+const data = [{
+  id: objectId,
+  bucketId: bucketId,
+  path: 'path',
+  public: 'true',
+  active: 'true',
+  createdBy: userId,
+  permCode: 'READ'
+}];
+
+const params = {
+  objId: objectId,
+  bucketId: bucketId,
+  createdBy: userId,
+  permCode: 'READ',
+  userId: userId
+};
+
+beforeEach(() => {
+  MockModel.mockReset();
+  MockTransaction.mockReset();
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+
+describe('addPermissions', () => {
+
+  it('should search and add object permissions', async () => {
+    // add userId property
+    const dataWithUser = { ...data, userId: userId };
```

Comment:
we dont want to do this. that was just to check the `if data.userId(...` logic in another function.
you can define the params you need within the scope of this test block.